torch/distributed/rpc/api.py

```diff
@@ -216,7 +216,8 @@ def _all_gather(obj, worker_names=None, timeout=UNSET_RPC_TIMEOUT):
 
     with _all_gather_dict_lock:
         states = _all_gather_sequence_id_to_states[sequence_id]
-    states.proceed_signal.wait()
+    timeout = timeout if timeout else None
+    states.proceed_signal.wait(timeout=timeout)
```

Comment:
Hey @H-Huang, just to double check. Would I be correct that the timeout here will be 0 ad it will make this call wait indefinitely, so that behavior does not change?


If so, could you please add a PR summary (and add to diffs as well) explaining that. Thanks!

I saw the following code:

```
    if timeout == UNSET_RPC_TIMEOUT:
        timeout = get_rpc_timeout()
```

Would I be correct that `get_rpc_timeout()` will return 0 by default?

Reply:
Thanks, for noticing this, I needed to fix some logic in the current implementation. Right now `UNSET_RPC_TIMEOUT = -1`,and `get_rpc_timeout()` gets whatever timeout is set by the RPC agent (default is 180).

Since we want to have keep the default behavior, of signal blocking indefinitely and RPC calls should keep their existing timeout. I added logic for two timeouts `signal_timeout` and `rpc_timeout`. https://github.com/pytorch/pytorch/pull/76194/files#diff-7f4f89fbe16f5d3e5e385d2625716f83a2e39477ec6c5d4cc51419bc0028b4c1R205-R211.

So the logic is now:
1. if timeout = UNSET_RPC_TIMEOU: RPC calls will default to the agent timeout, signal timeout will be infinite. This is the existing behavior.
2. if timeout is specified by user, RPC call and signal timeout will both use the same timeout. This is the added behavior.

Reply:
Makes sense to me! Thanks for the explanation!